contracts/interfaces/IV4Router.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import "forge-std/console.sol";
+import {Hooks} from "@uniswap/v4-core/contracts/libraries/Hooks.sol";
+import {IPoolManager} from "@uniswap/v4-core/contracts/interfaces/IPoolManager.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {TickMath} from "@uniswap/v4-core/contracts/libraries/TickMath.sol";
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+
+/// @title UniswapV4Routing
+/// @notice Abstract contract that contains all internal logic needed for routing through Uniswap V4 pools
+interface IV4Router {
+    error NotPoolManager();
+    error InvalidSwapType();
+    error TooLittleReceived();
+    error TooMuchRequested();
+
+    struct SwapInfo {
+        SwapType swapType;
+        address msgSender;
+        bytes params;
+    }
+
+    struct PathKey {
+        Currency tradeCurrency;
```

Comment:
I dont like the name `tradeCurrency`, its taken me a while to understand it. I guess I would call it like `intermediateCurrency`? Open to other ideas. just `tradeCurrency` wasn't immediately apparent to me

Reply:
great name, yah I just threw tradeCurrency in as a placeholder. I was hoping for something short to avoid annoying linting, but ultimately intermediateCurrency is a good option